opass

```diff
@@ -315,12 +315,17 @@ def main():
         # If it turns out that some people really like the tree-style
         # output, we can easily add it back as an option, controlled
         # by a flag, a subcommand, an environment variable, whatever.
-        pass_top = password_store_root()
-        for root, dirs, files in os.walk(pass_top):
-            for f in files:
-                if f.endswith(".gpg"):
-                    # Substring skips leading slash and chops trailing ".gpg".
-                    print('{}/{}'.format(root.replace(pass_top, ''), f)[1:-4])
+        pass_root = password_store_root()
+        for dirpath, _, filenames in os.walk(pass_root):
+            # subpath within password_store_root
+            dirname = dirpath.replace(pass_root, '')
+            for filename in filenames:
+                if filename.endswith(".gpg"):
+                    # Remove leading slash and trailing ".gpg"
+                    path = '{}/{}'.format(dirname, filename)[1:-4]
+                    if subpath and not path.startswith(subpath):
```

Comment:
This conditionalizes on *any* prefix, not necessarily a path-component-wise prefix.  In other words, if the user were to pass "far" as the argument, then the path "farmer/brown" would be considered a match.  Is this the desired behavior?

(Documenting the behavior of the new `subpath` parameter would have forced this question into consideration from the start, because to document it we'd have to decide what the precise behavior should be :-). )

Reply:
Yep, I considered this but didn't document it. I decided that a strict string prefix match was easier to implement than one that considered path components as atomic units. I'll document this, but if you think the other behavior is preferable and worth the additional effort (I don't think either `fnmatch` nor `glob` does exactly what we need), I can work on that.

Reply:
Renamed in 9793b32b5422cb673d4633844160c2dd085abbbf and documented in d15ee67e7aa7dea970c29ebf0c20e5bc81b5cd8a

Reply:
I think it's worth the additional effort -- it's really a correctness issue: the intended/desired behavior specifically path-component-wise not string-wise.  Also, I think the `os.path` library in Python has stuff to help with this?  See `os.path.commonprefix()` and other stuff in [os.path](https://docs.python.org/3/library/os.path.html).

Reply:
Ah, yep, the `os.path` module should work.

Reply:
Great!  Yeah, that module is full of useful tricks.

Reply:
Just double-checking that ball is still in your court, @gtback (for converting string-prefix behavior to pathwise-prefix behavior).  No rush; I just wanted to make sure I had a correct understanding of the current status of this PR.